sdk/core/azure-core/inc/http/policy.hpp

```diff
@@ -104,6 +109,9 @@ namespace Azure { namespace Core { namespace Http {
     std::unique_ptr<Response> Send(Context& ctx, Request& request, NextHttpPolicy nextHttpPolicy)
         const override
     {
+      auto uuid = UUID().GetUUIDString();
```

Comment:
I would also recommend the constructor to NOT create a random UUID, but follow the .NET pattern.
If you don't want to bother with the public constructors for now - that's fine, make them private, but I recommend to add a static method called `RandomUiid()` that basically does what the `UUID constructor` does now.

Reply:
Well, I also feel it's a little bit weird that we construct a `UUID` instance and then call `GetUUIDString` to get the real UUID. What else can one do with the `UUID` object other than call `GetUUIDString` and who wants to call `GetUUIDString` more than once?

But anyway, i can live with it as long as it works.